Added policy content for ANSSI minimal.

matejak opened the pull request:
This PR introduces the `anssi-minimal.yml` control definition that defines the  ~minimal~ minimal and intermediary ANSSI "security controls". The corresponding ANSSI profile files have been updated accordingly.

Observations:

- The policy format basically mirrors existing comments, which indicates that it is organic, and very likely a good design.
- I am torn regarding whether to fill in the Title/Description field in the policy file.
   - why not: It is the territory of OpenControl/OSCAL
   - why yes: ANSSI doesn't have OC/OSCAL content, and e.g. the OpenControl repository at https://github.com/ComplianceAsCode/redhat wasn't designed to handle something else than NIST 800-53. Presence of these metadata may make the policy file nicer, and it may make updates easier.
- I like the concept of "automated: partially", as it is a place where to declare to other SMEs that YMMV
- ANSSI Intermediary profile can simply extend ANSSI minimal, and add all controls of the `anssi-intermediary` TBD policy. This has a subtle flaw that the intermediary policy file will express only the diff between minimal and intermediary, but extending policies looks like a Pandora's box to me - we would probably have to handle extension and overrides of rule selections of individual controls, and that could be difficult to design in a good way.

matejak committed:
Added policy content for ANSSI minimal.

and updated the profile accordingly.

yuumasato commented on rhel7/profiles/anssi_nt28_minimal.profile:
With the "policy level" it would look like:
`- anssi:minimal`

matejak replied:
We don't have support for tags at the moment, so I have changed the approach to select individual controls. How does that look? The idea of introducing tags seems to fit here though, as we would probably prefer to have the "minimal" assignment in the policy file, not in the profile file.

yuumasato replied:
> We don't have support for tags at the moment, so I have changed the approach to select individual controls. How does that look?

Where? I don't see the changes you are talking about.

yuumasato commented on controls/anssi-minimal.yml:
```diff
@@ -0,0 +1,51 @@
+policy: 'ANSSI-BP-028-minimal'
```
To me it seems more natural to have a policy covering whole ANSSI.
I understand it as a "library" of requirements that any profile could source requirements from.

matejak replied:
I concur.

jan-cerny replied:
So do I.

yuumasato reviewed (commented):
Looks pretty good and promising.

>     * I am torn regarding whether to fill in the Title/Description field in the policy file.

I'm leaning to yes. At a least the title, it is hard to relate to a requirement that is just numbers.

> I like the concept of "automated: partially", as it is a place where to declare to other SMEs that YMMV

Is it more about automation or coverage? Example: what about a rule assigned to a control that won't ever have a remediation?
Or is it to represent that the control is open to interpretation and what is available in the system?

>     * ANSSI Intermediary profile can simply extend ANSSI minimal, and add all controls of the `anssi-intermediary` TBD policy.

Do the control files need to "align" with the profiles? I'd go simple and have a single control file for ANSSI.
`Minimal` and `Intermediary` could act as some kind of "policy level" selectors (I'm not sure if current implementation supports that). This is similar to `all` that was used in the profile.
I think that CIS policy would benefit from some kind "policy levels" too, in this case it would probably be `Level 1` or `Level 2`.
These policy selectors would help organize the controls and make it easy to define profiles aligned with these "levels".

matejak commented:
> Is it more about automation or coverage? Example: what about a rule assigned to a control that won't ever have a remediation?
Or is it to represent that the control is open to interpretation and what is available in the system?

The policy file addresses the control->rules mapping, so my intention was to express that the mapping itself may be questioned. You are also right - even if that mapping is solid, mapped rules may not check or fix anything.

I see your remark about "levels" or maybe generally tags, I will try to propose something. It's a great observation that multiple profiles may benefit from that.

When I get back to our recent CIS discussions, the added value of this could be to propagate the control ID to the rule's reference - right now, there is no enforcement between a reference and an actual selection.
I have noticed that the ANSSI reference has a format of `NT28(<control ID>)`. I think that formatting of references should be defined from the policy file as well - it is the only place that can, to some degree, point to what is NT28.

matejak committed:
Improved the ANSSI policy-profile integration.

Added the ANSSI minimal profile to the list of stable profiles.

jan-cerny commented on controls/anssi-minimal.yml:
```diff
@@ -0,0 +1,61 @@
+controls:
+  - id: R1
+    title: Minimization of installed services
+    automated: partially  # The list of essential services is not objective.
```
What about some `notes` key? Its contents could propagate to the generated outputs that we plan to introduce in future.

jan-cerny reviewed (commented):
This is really cool!

matejak committed:
Implemented concept of levels.

Policies, controls and profiles can have a level assigned to them.
Low-level controls are supposed to be essential and always be present in a profile,
while the higher-level controls may be used only in strict profiles.

matejak committed:
Ported the ANSSI intermediary profile to policies/controls.

jan-cerny commented on controls/anssi-minimal.yml:
```diff
@@ -3,59 +3,402 @@ title: 'ANSSI-BP-028'
+    # rules: TBD
+
+  - id: R12
+    level: intermediary
```
Do the levels include the lower levels? For example if I use `anssi:all:high` in my profile will the  intermediary control `R12` be included in that profile? If yes, how does it work?

matejak replied:
Yes, there is that method `get_all_controls_of_level_at_least`.

matejak committed:
Added support for notes in controls.

yuumasato committed:
Expand and discuss first 3 ANSSI requirements

yuumasato commented on controls/anssi-minimal.yml:
```diff
@@ -0,0 +1,404 @@
+policy: 'ANSSI-BP-028'
```
The policy file should be renamed to `controls/anssi.yml`.

yuumasato replied:
But only after https://github.com/matejak/scap-security-guide/pull/5 is merged, :grin:

matejak committed:
Merge pull request #5 from yuumasato/extend_3_first_anssi_reqs

Expand and discuss first 3 ANSSI requirements

matejak committed:
Renamed ANSSI policy file to more appropriate anssi.yml

jan-cerny commented on controls/anssi.yml:
```diff
@@ -0,0 +1,442 @@
+policy: 'ANSSI-BP-028'
```
Nice catch! I confirm using BP is correct for the latest version. The profile files should be renamed, then.

jan-cerny reviewed (commented):
@matejak This is perfect.

I only think that the profile files need to be rename.

Do you want to merge this into `profile_with_inline_policies` branch? In other words, do you prefer the inline policies variant over profiles with separate policies?

matejak committed:
Updated the profile titles from NT to BP

NT refers to the outdated version 1.1, whereas BP is for 1.2.
There is no need to target 1.1

matejak commented:
I have done some research about Debian-based profiles, and they are mostly the same across Debian/Ubuntu versions, but they substantially differ from the RHEL7 profile.
Those profile files don't contain comments, and they are from the before-refactoring time when there were no `.profile` files at all. So they probably target an older version of ANSSI, and there is no way for us to integrate them into this policy file without a thorough analysis.

matejak committed:
Increase controls handling robustness.

Don't parse other files than those that end by .yml

yuumasato committed:
Discuss and expand selection of rules for ANSSI R5

ANSSI R5 is about defense in-depth, which is broad topic.
Note a few aspect of defense in-depth and add a few related rules.

jan-cerny commented:
This is great!
